### ISSUE TITLE
2 improvement things

### ISSUE THREAD

**Issue by kellerkichel** [filamentgroup/loadCSS#108]
I have make some tests... and after a time I have 2 new improvement to the script, following code:

The first one stop the load function when the style sheets is already on page.

    ...
    w.loadCSS = function (href, before, media) {
        
        // stop placing duplacated styleSheets
        var ss_loaded = document.styleSheets;
        for (var i = ss_loaded.length - 1; i >= 0; i--) {
            if (ss_loaded[i].href.indexOf(href) > -1) return false;
        }
    ...

And the second is a redraw for Chrome 45.0.2454.101, sometimes it makes the callback before the css media is set to all, so a reflow on page normalize it.

    ...
    onloadcssdefined(function () {
        ss.media = media || "all";
    
        // force a reflow (issue on chrome some times it dont refresh the layouy until we move mouse or something)
            ss.style.display = 'block';
            ss.style.display = 'none';
        });
    ...

I hope this help someone

*Note: I use these with [requirejs](http://requirejs.org/)*

**Comment by scottjehl** [comment]
This looks really great!
Thanks so much. We're pretty swamped this week with project work but we'll try to get this tested and in soon

**Comment by kellerkichel** [comment]
I have make some improvements... 

the reflow part its useless... and the check need another check for src:

    var ss_loaded = document.styleSheets;
    for (var i = ss_loaded.length - 1; i >= 0; i--) {
        if (ss_loaded[i].href) {
            if (ss_loaded[i].href.indexOf(href) > -1) return false;
        }
    }

I dont know if is the better way.. but works :+1:

**Comment by wouterbulten** [comment]
@kellerkichel Your new snippet solved the double load problem for me (related to #110). Thanks :+1:

**Comment by scottjehl** [comment]
So, I just reviewed this and we really appreciate it. Unfortunately, while it does prevent the double request in Chrome, it comes at a couple of costs we don't want to introduce. Triggering layout is one of those costs, and while it may not visibly do anything undesirable, setting a link to display: block, even temporarily is at least unusual, and may be wise to avoid. Also, there may be cases where folks do actually want to load the same url more than once, such as when caching headers are set to expire immediately. 

Fortunately, I've got a simple workaround on the way that addresses this bug in a simple manner. Will post today. Thanks again!
